routing/RouteSolver.py

```diff
@@ -0,0 +1,129 @@
+
+def get_graph_details_from_user():
+    number_of_nodes = 0;
+    while True:
+        try:
+            number_of_nodes = int(input("How many nodes are in your graph? "))
+            break
+        except ValueError:
+            print('Invalid input. Please input an integer')
+            continue
+    node_count = 0
+    while node_count < int(number_of_nodes):
+        new_name = input("Please enter a name for node {node_count}\n")
+        graph.add_vertex(new_name)
+        node_count += 1
+    print(graph.vertices())
+    for node in graph.vertices():
+        another_node = True
+        while another_node:
+            node_name = input("Please enter connected node name for node {node}:\n")
+            if node_name in graph.vertices():
+                while True:
+                    direction = input("Please enter the cardinal direction traveling from {node} to {node_name}: {DIRECTIONS}\n")
+                    if direction in DIRECTIONS:
+                        graph.add_edge((node, node_name))
+                        another_connection = bool_input("Is there another connected node?")
+                        if another_connection:
+                            another_node = True
+                        else :
+                            another_node = False
+                        break
+                    else:
+                        print('Invalid direction given. Valid directions ({DIRECTIONS})')
+            else:
+                connected = bool_input("Invalid node name given. Is there a node connected?")
+                if connected:
+                    another_node = True
+                else:
+                    another_node = False
+
+def main(args):
+    if args.c:
+        load_config(args.c)
+    else:
+        get_graph_details_from_user()
+    while True:
+        start_node = input("Please enter the start node:\n")
+        if start_node in graph.vertices() :
+            break;
+        else:
+            print("Invalid start node provided\n")
+    while True:
+        end_node = input("Please enter the end node:\n")
+        if end_node in graph.vertices():
+            break;
+        else:
+            print("Invalid end node provided\n")
+    paths = graph.find_path(start_node, end_node)
+    print("This is a single path: ")
+    print(paths)
+    all_paths = graph.find_all_paths(start_node, end_node)
+    print("These are all possible paths: ")
+    print(all_paths)
+    for p in all_paths:
+        print(p)
+
+##################################################################################
+##                                                                              ##
+##                            Exeutable code below                              ##
+##                                                                              ##
+##################################################################################
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
```

Comment:
Move this into the `main` function. No point in having it outside of it. Should be...

```python
if __name__ == '__main__':
    main()
```

Reply:
On the argument have both `-c` and `--config`